pkg/api/login.go

```diff
@@ -81,6 +81,13 @@ func (hs *HTTPServer) LoginView(c *models.ReqContext) {
 		return
 	}
 
+	urlParams := c.Req.URL.Query()
+	if _, isInternal := urlParams["internal"]; isInternal {
+		hs.log.Info("Using internal login, skipping OAuth auto login")
```

Comment:
```suggestion
		hs.log.Debug("Redirect manually overridden, skipping auto login")
```

Mostly Info->Debug, since this isn't really of interest to the server admin unless something is wonky, but also generalized the language a little. You can still manually click the OAuth provider when you force not redirecting, tried to reflect that in the new language.